src/validation.cpp

```diff
@@ -3792,6 +3794,21 @@ void BlockManager::FindFilesToPrune(std::set<int>& setFilesToPrune, uint64_t nPr
            nLastBlockWeCanPrune, count);
 }
 
+void BlockManager::UpdatePruneBlocker(const std::string& name, const CBlockIndex* block) {
+    AssertLockHeld(::cs_main);
+    m_prune_blockers[name] = block;
+}
+
+void BlockManager::GetLastPruneBlockerHeight(int& last_prune) {
+    for (auto const& blocker : m_prune_blockers) {
+        const int blocker_height = blocker.second->nHeight + PRUNE_BLOCKER_BUFFER;
+        last_prune = std::max(1, std::min(last_prune, blocker_height));
+        if (last_prune == blocker_height) {
+            LogPrintf("%s limited pruning to height %d\n", blocker.first, blocker_height);
```

Comment:
In commit "Add prune blockers to BlockManager" (9f8d6a9a8f7d113ae12306c72a773830b5725e37):

Would it make sense to use a log category here `LogPrint(BCLog::PRUNE, ...)` instead of printing unconditionally? Also in the future it could be nice to unify this print with the prints in the FindFilesToPrune functions and print the full pruning status in a consistent way on just one line.

Reply:
Yeah, I agree, I added a log category. I am leaving unifying for a potential follow-up to not complicate this PR further.